android/src/main/java/com/auth0/react/A0Auth0Module.java

```diff
@@ -122,29 +123,35 @@ String generateCodeChallenge(@NonNull String codeVerifier) {
     }
 
     @Override
-    public void onHostResume() {
-        new Handler().postDelayed(new Runnable() {
+    public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent data) {
+        new Handler().post(new Runnable() {
             @Override
             public void run() {
                 Callback cb = A0Auth0Module.this.callback;
+
                 if (cb != null) {
-                    final WritableMap error = Arguments.createMap();
-                    error.putString("error", "a0.session.user_cancelled");
-                    error.putString("error_description", "User cancelled the Auth");
-                    cb.invoke(error);
+                    if (resultCode == RESULT_OK &&
```

Comment:
can we extract the 3 conditions into a single variable/flag with a self-describing name?

Reply:
Done in https://github.com/auth0/react-native-auth0/pull/350/commits/4b7998584f638b1b0909a90a37f7a57967121b53.